contracts/dydx-airdrop/src/build_messages.rs

```diff
@@ -0,0 +1,140 @@
+use cosmwasm_std::{
+    coins, to_json_binary, Addr, BankMsg, Deps, DepsMut, Env, MessageInfo, StdResult, WasmMsg,
+};
+use sg_std::{CosmosMsg, Response, StargazeMsgWrapper, SubMsg, NATIVE_DENOM};
+use whitelist_updatable::msg::ExecuteMsg;
+// TODO: Replace with whitelist_updatable_flatrate
+use crate::msg::InstantiateMsg;
+use crate::query::{query_collection_whitelist, query_mint_count};
+use crate::state::{Config, CONFIG};
+use crate::ContractError;
+use sg_whitelist_flex::helpers::interface::CollectionWhitelistContract;
+use sg_whitelist_flex::msg::AddMembersMsg;
+use sg_whitelist_flex::msg::{ExecuteMsg as CollectionWhitelistExecuteMsg, Member};
+use whitelist_immutable_flex::msg::InstantiateMsg as WIFInstantiateMsg;
+use whitelist_updatable::msg::ExecuteMsg::AddAddresses;
+pub const IMMUTABLE_WHITELIST_LABEL: &str = "Whitelist Immutable Flex for Airdrop";
+pub const INIT_WHITELIST_REPLY_ID: u64 = 1;
+
+pub fn whitelist_instantiate(
+    env: Env,
+    msg: InstantiateMsg,
+) -> Result<cosmwasm_std::SubMsg<StargazeMsgWrapper>, ContractError> {
+    let whitelist_instantiate_msg = WIFInstantiateMsg {
+        members: msg.members,
+        mint_discount_bps: Some(0),
+    };
+    let wasm_msg = WasmMsg::Instantiate {
+        code_id: msg.whitelist_code_id,
+        admin: Some(env.contract.address.to_string()),
+        funds: vec![],
+        label: IMMUTABLE_WHITELIST_LABEL.to_string(),
+        msg: to_json_binary(&whitelist_instantiate_msg)?,
+    };
+    Ok(SubMsg::reply_on_success(wasm_msg, INIT_WHITELIST_REPLY_ID))
+}
+
+pub fn state_config(
+    deps: Deps,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Config, ContractError> {
+    Ok(Config {
+        admin: info.sender,
+        claim_msg_plaintext: msg.clone().claim_msg_plaintext,
+        airdrop_amount: crate::validation::validate_airdrop_amount(msg.airdrop_amount)?,
+        whitelist_address: None,
+        minter_address: deps.api.addr_validate(msg.minter_address.as_ref())?,
+        name_discount_wl_address: deps
+            .api
+            .addr_validate(msg.name_discount_wl_address.as_ref())?,
+        name_collection_address: deps
+            .api
+            .addr_validate(msg.name_collection_address.as_ref())?,
+        airdrop_count_limit: msg.airdrop_count_limit,
+    })
+}
+
+pub fn claim_reward(info: MessageInfo, airdrop_amount: u128) -> Result<Response, ContractError> {
+    let mut res = Response::new();
+    let bank_msg = SubMsg::new(BankMsg::Send {
+        to_address: info.sender.to_string(),
+        amount: coins(airdrop_amount, NATIVE_DENOM),
+    });
+    res = res.add_submessage(bank_msg);
+
+    Ok(res)
+}
+
+pub fn dust_and_whitelist_add(
```

Comment:
I think this can just be whitelist_add and call the dust( ) function from the calling function